CHANGELOG.md

```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 2.18
```

Comment:
I would prefer to see this as a major version number, and force commands to be run within DDEV. We should not be using `git` within the DDEV environment, but only on the host. See also OTDEVOPS-309.

Reply:
I don't think this change will run git in ddev but it will run the commands that grumphp runs inside ddev.

However, I am curious if this would still require php to be present on the host? @youwe-petervanderwal 

In one of my projects, made a change locally that makes the git hook run grumphp itself in ddev instead of just the commands grumphp is running.

Reply:
In my opinion this is not a breaking change so a minor version is enough. Making it a next major version would require developers to explicitly upgrade to the new version while using a minor version would apply this update by running `composer update` (although you might still need to run `ddev exec vendor/bin/grumphp git:init`...)

What do others think?

I personally don't want to prohibit users to use git in DDev as using git in DDev itself is not the problem. The problem is "DDEV User" as author names (which should be checked/restricted via another route if you ask me).

Reply:
I also don't see why this would be a major update:

- Will there be issues if we keep this a minor update? 
- Will certain functionality no longer be compatible, and thus prefering a major update?

Reply:
> I don't think this change will run git in ddev but it will run the commands that grumphp runs inside ddev.

Indeed - you can (probably should) run your `git commit` command outside DDev. Which is now not possible unless you have the correct PHP version installed locally.

> However, I am curious if this would still require php to be present on the host? @youwe-petervanderwal
> 
> In one of my projects, made a change locally that makes the git hook run grumphp itself in ddev instead of just the commands grumphp is running.

To answer your question: no, you don't need PHP on your host anymore. In combination with the config change in [`default/grumphp.yml`](config/default/grumphp.yml) the Git hook will be (re)written as

```
# cat .git/hooks/pre-commit

# Run GrumPHP
(cd "./" && printf "%s\n" "${DIFF}" | vendor/bin/youwe-ddev-detected-exec 'vendor/bin/grumphp' 'git:pre-commit' '--skip-success-output')
```

The default is `... exec 'vendor/bin/grumphp'` which will run Grump on the same container as you're executing `git commit`.

If you want to run Grump manually, you have to remember yourself to execute it with `ddev exec vendor/bin/grumphp` (or within `ddev ssh`) -- but that's the same with a lot of other commands.